numba/core/callconv.py

```diff
@@ -684,7 +684,10 @@ def set_dynamic_user_exc(self, builder, exc, exc_args, nb_types, loc=None,
 
         # serialize comp. time args
         pyapi = self.context.get_python_api(builder)
-        exc = self.build_excinfo_struct(exc, exc_args, loc, func_name)
+        dummy = self.context.get_dummy_value()
+        exc_args_static = tuple(
+            [dummy if isinstance(arg, ir.Value) else arg for arg in exc_args])
+        exc = self.build_excinfo_struct(exc, exc_args_static, loc, func_name)
```

Comment:
Please could a test be added to show what was failing that is now fixed via this change? Many thanks.